integration_tests/integration_tests/tests/dbt_project.py

```diff
@@ -98,15 +142,32 @@ def test(
                 YAML().dump(props_yaml, props_file)
                 relative_props_path = Path(props_file.name).relative_to(PATH)
                 self.dbt_runner.test(select=str(relative_props_path))
-        return self._read_test_result(test_id)
+
+        if multiple_results:
+            return self._read_test_results(test_id)
+        else:
+            return self._read_single_test_result(test_id)
 
     def seed(self, data: List[dict], table_name: str):
         return DbtDataSeeder(self.dbt_runner).seed(data, table_name)
 
-    def _read_test_result(self, table_name: str) -> Dict[str, Any]:
-        return self.read_table(
+    def _read_test_results(self, table_name: str) -> List[Dict[str, Any]]:
+        test_execution_id = self.read_table(
             "elementary_test_results",
             where=f"lower(table_name) = lower('{table_name}')",
             order_by="created_at DESC",
+            column_names=["test_execution_id"],
             limit=1,
-        )[0]
+        )[0]["test_execution_id"]
+        return self.read_table(
+            "elementary_test_results",
+            where=f"test_execution_id = '{test_execution_id}'",
+        )
+
+    def _read_single_test_result(self, table_name: str) -> Dict[str, Any]:
```

Comment:
Personally don't think the `single` is really necessary given that it's `result` and not `results`.

Reply:
It's not a must but I think it is a bit more explicit